src/annotator/config/index.js

```diff
@@ -60,71 +68,100 @@ function configurationKeys(appContext) {
-  const resultConfig = {};
-  settingsKeys.forEach(key => {
-    resultConfig[key] = allConfigSettings[key];
-  });
-  return resultConfig;
-}
+const configDefinitions = {
+  annotations: {
+    getValue: settings => settings.annotations,
+  },
+  appType: {
+    getValue: settings =>
+      settings.hostPageSetting('appType', {
+        allowInBrowserExt: true,
+      }),
+  },
+  branding: {
+    getValue: settings => settings.hostPageSetting('branding'),
+  },
+  // URL of the client's boot script. Used when injecting the client into
+  // child iframes.
+  clientUrl: {
+    getValue: settings => settings.clientUrl,
+  },
+  enableExperimentalNewNoteButton: {
+    getValue: settings =>
+      settings.hostPageSetting('enableExperimentalNewNoteButton'),
+  },
+  group: {
+    getValue: settings => settings.group,
+  },
+  focus: {
+    getValue: settings => settings.hostPageSetting('focus'),
+  },
+  theme: {
+    getValue: settings => settings.hostPageSetting('theme'),
+  },
+  usernameUrl: {
+    getValue: settings => settings.hostPageSetting('usernameUrl'),
+  },
+  onLayoutChange: {
+    getValue: settings => settings.hostPageSetting('onLayoutChange'),
+  },
+  openSidebar: {
+    getValue: settings =>
+      settings.hostPageSetting('openSidebar', {
+        allowInBrowserExt: true,
+        coerce: toBoolean,
+      }),
+  },
+  query: {
+    getValue: settings => settings.query,
+  },
+  requestConfigFromFrame: {
+    getValue: settings => settings.hostPageSetting('requestConfigFromFrame'),
+  },
+  services: {
+    getValue: settings => settings.hostPageSetting('services'),
+  },
+  showHighlights: {
+    getValue: settings => settings.showHighlights,
+  },
+  notebookAppUrl: {
+    getValue: settings => settings.notebookAppUrl,
+  },
+  sidebarAppUrl: {
+    getValue: settings => settings.sidebarAppUrl,
+  },
+  // Sub-frame identifier given when a frame is being embedded into
+  // by a top level client
+  subFrameIdentifier: {
+    getValue: settings =>
+      settings.hostPageSetting('subFrameIdentifier', {
+        allowInBrowserExt: true,
+      }),
+  },
+  externalContainerSelector: {
+    getValue: settings => settings.hostPageSetting('externalContainerSelector'),
+  },
+};
 
 /**
  * Return the configuration for a given application context.
  *
  * @param {AppContext} [appContext] - The name of the app.
  */
 export function getConfig(appContext = 'annotator', window_ = window) {
+  const settings = settingsFrom(window_);
+  const config = {};
   // Filter the config based on the application context as some config values
   // may be inappropriate or erroneous for some applications.
-  const filteredKeys = configurationKeys(appContext);
-  const config = configFrom(filteredKeys, window_);
+  let filteredKeys = configurationKeys(appContext);
+  filteredKeys.forEach(name => {
+    const configDef = configDefinitions[name];
+    // Get the value from the configuration source and run through an optional coerce method
```

Comment:
```suggestion
    // Get the value from the configuration source
```

There is no coercion happening currently. Presumably this will come in future.

Reply:
I believe some of these comments were merge errors dude to upstream changes before. This might be one of them, good catch.